### PR TITLE
Wallet Backup Initial Screen

### PR DESCRIPTION
**Pull request by yhaspel**
Starting work on wallet backup.
This commit only includes route to empty backup page when clicking on the backup wallet link from wallet overview page.

Screen Shot 2019-04-01 at 2 35 43 PM

### CODE REVIEW

File: app/screens/wallet/Backup.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        <ActionLink onClick={this.learnMoreAboutSecurity}>Learn more about wallet security and backup</ActionLink>
+      </Wrapper>
+    );
+  }
+
+  renderBackupBox = (mode: 'file' | 'mnemonic') => {
+    const shieldLogo = mode === 'file' ? shieldIconGreenOne : shieldIconOrangeTwo;
+    const securityLevel = `${mode === 'file' ? 'Basic' : 'Enhanced'} Security`;
+    const backupMode = `${mode === 'file' ? 'File' : '12 Words'} Backup`;
+    const bodyText = {
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: extract to const outside of class

File: app/screens/wallet/Backup.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      file: 'Take your wallet file and save a copy of it in a secure location - We recommend a USB key or a external drive.',
+      mnemonic: 'A list of words you keep on a physical paper or in your password manager, that can be used to recover your wallet on any device.'
+    };
+    const noteText = {
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: same as above

File: app/screens/wallet/Backup.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      file: 'NOTE: You will still need your passphrase to restore the wallet from the file.',
+      mnemonic: 'NOTE: Works even if you lost your passphrase or wallet backup file'
+    };
+    const linkText = mode === 'file' ? 'Show me the wallet file' : 'Create a paper backup';
+    const linkActions = {
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: why object and not pointer to func straight forward?

File: app/assets/images/index.js [hunk 4]
```diff
@@ -25,6 +25,8 @@ export { default as miner } from './mining.gif';
 export { default as welcomeBack } from './welcome_back@2x.png';
 export { default as copyIcon } from './copy_green@2x.png';
 export { default as shieldIconGreen } from './security_logo_green@2x.png';
+export { default as shieldIconGreenOne } from './security_logo_green_one@2x.png';
```
Comment (IlyaVi) [inline app/assets/images/index.js]: there is other shieldIcon? If not - rename to shieldIconGreen

File: app/assets/images/index.js [hunk 5]
```diff
@@ -25,6 +25,8 @@ export { default as miner } from './mining.gif';
...
+export { default as shieldIconOrangeTwo } from './security_logo_orange_two@2x.png';
```
Comment (IlyaVi) [inline app/assets/images/index.js]: same as above

File: app/screens/wallet/Backup.js [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+          </BackupTitleWrapper>
+          {mode === 'mnemonic' && (
+            <Recommended>
+              <GrayText>Recommended</GrayText>
+            </Recommended>
+          )}
+        </BackupTopWrapper>
+        <BaseText>{bodyText[mode]}</BaseText>
+        <BaseText>{noteText[mode]}</BaseText>
+        <ActionLink onClick={linkActions[mode]} theme={mode === 'file' ? 'green' : 'orange'}>
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: why not defining color here and making another evaluation in styled component?

File: app/screens/wallet/Backup.js [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+          {this.renderBackupBox('file')}
+          {this.renderBackupBox('mnemonic')}
+        </BackupBoxesWrapper>
+        <ActionLink onClick={this.learnMoreAboutSecurity}>Learn more about wallet security and backup</ActionLink>
+      </Wrapper>
+    );
+  }
+
+  renderBackupBox = (mode: 'file' | 'mnemonic') => {
+    const shieldLogo = mode === 'file' ? shieldIconGreenOne : shieldIconOrangeTwo;
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: same evaluation is repeated multiple time

File: app/screens/wallet/Backup.js [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+  cursor: pointer;
+  ${Actionable}
+`;
+
+// $FlowStyledIssue
+const BackupBox = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  ${({ borderColor }) =>
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: border color is provided in any case - simplify

File: app/screens/wallet/Backup.js [hunk 9]
```diff
@@ -0,0 +1,187 @@
+// @flow
+import React, { Component } from 'react';
+import styled, { css } from 'styled-components';
+import { smColors } from '/vars';
+import { shieldIconGreenOne, shieldIconOrangeTwo } from '/assets/images';
+
+const Actionable = css`
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: this component is used only once

File: app/assets/images/index.js (hunk 4 above)
Comment (yhaspel) [reply app/assets/images/index.js]: Yes. the regular shield icon is not numbered. this one has "1", the other one has a "!" in its center.

File: app/assets/images/index.js (hunk 5 above)
Comment (yhaspel) [reply app/assets/images/index.js]: They are numbered, but in this case there is no other orange one. You still want this renamed?

File: app/screens/wallet/Backup.js (hunk 6 above)
Comment (yhaspel) [reply app/screens/wallet/Backup.js]: ok

File: app/screens/wallet/Backup.js (hunk 7 above)
Comment (yhaspel) [reply app/screens/wallet/Backup.js]: assigned to const

File: app/screens/wallet/Backup.js (hunk 8 above)
Comment (yhaspel) [reply app/screens/wallet/Backup.js]: ok

File: app/screens/wallet/Backup.js (hunk 9 above)
Comment (yhaspel) [reply app/screens/wallet/Backup.js]: integrated with ActionLink styled component

File: app/screens/wallet/Backup.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+        <ActionLink onClick={this.learnMoreAboutSecurity} color={smColors.green}>
+          Learn more about wallet security and backup
+        </ActionLink>
+      </Wrapper>
+    );
+  }
+
+  renderBackupBox = (mode: 'file' | 'mnemonic') => {
+    const isFileMode = mode === 'file';
+    const shieldLogo = isFileMode ? shieldIconGreenOne : shieldIconOrangeTwo;
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: shieldLogo, securityLevel, backupMode, linkText - all of these are constants and don't depend on any external source of truth. Extract them to constant outside of class and unite with bodyText/noteText

Comment (yhaspel) [reply app/screens/wallet/Backup.js]: ok

**Pull request merged**